src/lib.rs

```diff
@@ -1,5 +1,13 @@
-#![deny(missing_docs)]
 #![doc = include_str!("../README.md")]
+#![deny(missing_docs)]
+#![warn(clippy::all, clippy::pedantic)]
+#![allow(
+    clippy::cast_precision_loss,
+    clippy::float_cmp, // TODO: this should not be set globally
```

Comment:
We can allow that for tests only. I compare floats in tests (especially in doctests, so that user-facing examples show a precise result), not in the algorithms.